client/activities/activity-button.tsx

```diff
@@ -125,7 +125,7 @@ export interface ActivityButtonProps {
 }
 
 export const ActivityButton = React.memo(
-  React.forwardRef<HTMLButtonElement, ActivityButtonProps>(
+  React.forwardRef<HTMLButtonElement | undefined, ActivityButtonProps>(
```

Comment:
This is the wrong change to make. React uses a `null` initial ref value to indicate that it will completely control the ref (and that the ref is not mutable by outside APIs). The types also encode this.

The right fix here is to initialize the refs you're passing into this with `null`